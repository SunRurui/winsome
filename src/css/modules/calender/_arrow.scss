.arrowRight {
	//Prameters are left,top,height,width,rotationDegree
	@include arrow(92.4%, 67%, 40px, 40px, 135deg);
	border-left: 3px solid $arrowColor;
	border-top: 3px solid $arrowColor;
	transition: all 0.4s ease-in-out;
	&:active {
		transform: scale(1.2) rotate(135deg);
		border-right: transparent;
		border-bottom: transparent;
	}
	&:focus {
		border-left: 5px solid $arrowColor;
		border-top: 5px solid $arrowColor;
		border-right: transparent;
		border-bottom: transparent;
	}
}
.arrowLeft {
	//Prameters are left,top,height,width,rotationDegree
	@include arrow(4.5%, 67%, 40px, 40px, -45deg);
	border-left: 3px solid $arrowColor;
	border-top: 3px solid $arrowColor;
	transition: all 0.4s ease-in-out;
	&:active {
		transform: scale(1.2) rotate(-45deg);
		border-right: transparent;
		border-bottom: transparent;
	}
	&:focus {
		border-left: 5px solid $arrowColor;
		border-top: 5px solid $arrowColor;
		border-right: transparent;
		border-bottom: transparent;
	}
}
