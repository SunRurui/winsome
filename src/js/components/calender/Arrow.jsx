import React from 'react';
import { func } from 'prop-types';
import { connect } from 'react-redux';
import * as actions from '../../actions';

export const Arrow = ({ onClickRightArrow, onClickLeftArrow }) => (
	<div>
		<div
			className="arrowRight"
			onClick={onClickRightArrow}
			role="button"
			tabIndex="0"
		/>
		<div
			className="arrowLeft"
			onClick={onClickLeftArrow}
			role="button"
			tabIndex="0"
		/>
	</div>
);

Arrow.propTypes = {
	onClickRightArrow: func.isRequired,
	onClickLeftArrow: func.isRequired
};
export default connect(null, {
	onClickLeftArrow: actions.onClickLeftArrow,
	onClickRightArrow: actions.onClickRightArrow
})(Arrow);
