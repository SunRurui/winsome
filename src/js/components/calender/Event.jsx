import React from 'react'
import {connect} from 'react-redux'


export const Event=({counter,events})=>{
		console.log('counter  in events',counter)
		return [...events].slice(counter,counter+4).map(({...item},index)=>(
				<div key={index} className="event">
					<div className="comingEvents">
						{item['event']}
					</div>
					<div className="comingEventsDates">
							{item['eventDate']}
					</div>
					<div className="time">
							{item['time']}
					</div>
					<div className="clear"></div>
				</div>
				

		))
	}
function mapStateToProps({calender}) {
	return {
		counter:calender.counter,
		events:calender.events
	}
}
export default connect(mapStateToProps,null)(Event)
