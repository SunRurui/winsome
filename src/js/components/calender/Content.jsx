import React from 'react';
import Arrow from './Arrow';
import Event from './Event';
const Content = () => (
	<div className="heroImage">
		<div className="line" />
		<h2>
			<strong>Up Coming Events</strong>
		</h2>
		<div className="renderEvents">
			<Event />
		</div>
		<div className="clear" />
		<Arrow />
	</div>
);

export default Content;
