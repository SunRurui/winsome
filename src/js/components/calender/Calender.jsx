import React from 'react';
import Content from './Content';
import {connect} from 'react-redux';

export default class Calender extends React.Component {
	constructor(){
		super();
	}
	render(){
		
		return (
			<div>
			<Content/>	
			</div>
		)
	}
}