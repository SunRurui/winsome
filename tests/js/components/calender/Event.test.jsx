import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {Event} from '../../../../src/js/components/calender/Event';

Enzyme.configure({adapter:new Adapter()})

describe('Test the Event Component of Calender',()=>{

	let items=[{event:'Half terms End',eventDate:'01 May',time:''},{},{}];
	let counter=0;
	const wrapper=shallow(<Event counter={counter} items={items} />);
	it('checks the rendring',()=>{
		expect(wrapper).toMatchSnapshot();
	})
	
})
