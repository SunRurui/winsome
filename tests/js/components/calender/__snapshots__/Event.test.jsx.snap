// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Event Component of Calender checks the rendring 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Event
    counter={0}
    events={
      Array [
        Object {
          "event": "Half terms End",
          "eventDate": "01 May",
          "id": 1,
          "time": "",
        },
        Object {
          "event": "NetBall: first vii vs viii",
          "eventDate": "20 May",
          "id": 2,
          "time": "2:30 pm",
        },
        Object {
          "event": "Parent presentation on healthy living",
          "eventDate": "03 June",
          "id": 3,
          "time": "12:35 pm",
        },
        Object {
          "event": "M5 rs trip to walsigham",
          "eventDate": "25 June",
          "id": 4,
          "time": "",
        },
        Object {
          "event": "Tour to muree",
          "eventDate": "15 July",
          "id": 5,
          "time": "2:30 pm",
        },
        Object {
          "event": "Water splash Day",
          "eventDate": "28 July",
          "id": 6,
          "time": "11:35 am",
        },
        Object {
          "event": "Visit to Jungle",
          "eventDate": "05 Aug",
          "id": 7,
          "time": "",
        },
        Object {
          "event": "Independence Day",
          "eventDate": "14 Aug",
          "id": 8,
          "time": "12:35 am",
        },
        Object {
          "event": "Bsant Day",
          "eventDate": "25 Aug",
          "id": 9,
          "time": "11:38 pm",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="event"
        >
          <div
            className="comingEvents"
          >
            Half terms End
          </div>
          <div
            className="comingEventsDates"
          >
            01 May
          </div>
          <div
            className="time"
          >
            
          </div>
          <div
            className="clear"
          />
        </div>,
        <div
          className="event"
        >
          <div
            className="comingEvents"
          >
            NetBall: first vii vs viii
          </div>
          <div
            className="comingEventsDates"
          >
            20 May
          </div>
          <div
            className="time"
          >
            2:30 pm
          </div>
          <div
            className="clear"
          />
        </div>,
        <div
          className="event"
        >
          <div
            className="comingEvents"
          >
            Parent presentation on healthy living
          </div>
          <div
            className="comingEventsDates"
          >
            03 June
          </div>
          <div
            className="time"
          >
            12:35 pm
          </div>
          <div
            className="clear"
          />
        </div>,
        <div
          className="event"
        >
          <div
            className="comingEvents"
          >
            M5 rs trip to walsigham
          </div>
          <div
            className="comingEventsDates"
          >
            25 June
          </div>
          <div
            className="time"
          >
            
          </div>
          <div
            className="clear"
          />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="comingEvents"
            >
              Half terms End
            </div>,
            <div
              className="comingEventsDates"
            >
              01 May
            </div>,
            <div
              className="time"
            >
              
            </div>,
            <div
              className="clear"
            />,
          ],
          "className": "event",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Half terms End",
              "className": "comingEvents",
            },
            "ref": null,
            "rendered": "Half terms End",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "01 May",
              "className": "comingEventsDates",
            },
            "ref": null,
            "rendered": "01 May",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "time",
            },
            "ref": null,
            "rendered": "",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "clear",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="comingEvents"
            >
              NetBall: first vii vs viii
            </div>,
            <div
              className="comingEventsDates"
            >
              20 May
            </div>,
            <div
              className="time"
            >
              2:30 pm
            </div>,
            <div
              className="clear"
            />,
          ],
          "className": "event",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "NetBall: first vii vs viii",
              "className": "comingEvents",
            },
            "ref": null,
            "rendered": "NetBall: first vii vs viii",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "20 May",
              "className": "comingEventsDates",
            },
            "ref": null,
            "rendered": "20 May",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "2:30 pm",
              "className": "time",
            },
            "ref": null,
            "rendered": "2:30 pm",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "clear",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="comingEvents"
            >
              Parent presentation on healthy living
            </div>,
            <div
              className="comingEventsDates"
            >
              03 June
            </div>,
            <div
              className="time"
            >
              12:35 pm
            </div>,
            <div
              className="clear"
            />,
          ],
          "className": "event",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Parent presentation on healthy living",
              "className": "comingEvents",
            },
            "ref": null,
            "rendered": "Parent presentation on healthy living",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "03 June",
              "className": "comingEventsDates",
            },
            "ref": null,
            "rendered": "03 June",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "12:35 pm",
              "className": "time",
            },
            "ref": null,
            "rendered": "12:35 pm",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "clear",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="comingEvents"
            >
              M5 rs trip to walsigham
            </div>,
            <div
              className="comingEventsDates"
            >
              25 June
            </div>,
            <div
              className="time"
            >
              
            </div>,
            <div
              className="clear"
            />,
          ],
          "className": "event",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "M5 rs trip to walsigham",
              "className": "comingEvents",
            },
            "ref": null,
            "rendered": "M5 rs trip to walsigham",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "25 June",
              "className": "comingEventsDates",
            },
            "ref": null,
            "rendered": "25 June",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "time",
            },
            "ref": null,
            "rendered": "",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "clear",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="event"
          >
            <div
              className="comingEvents"
            >
              Half terms End
            </div>
            <div
              className="comingEventsDates"
            >
              01 May
            </div>
            <div
              className="time"
            >
              
            </div>
            <div
              className="clear"
            />
          </div>,
          <div
            className="event"
          >
            <div
              className="comingEvents"
            >
              NetBall: first vii vs viii
            </div>
            <div
              className="comingEventsDates"
            >
              20 May
            </div>
            <div
              className="time"
            >
              2:30 pm
            </div>
            <div
              className="clear"
            />
          </div>,
          <div
            className="event"
          >
            <div
              className="comingEvents"
            >
              Parent presentation on healthy living
            </div>
            <div
              className="comingEventsDates"
            >
              03 June
            </div>
            <div
              className="time"
            >
              12:35 pm
            </div>
            <div
              className="clear"
            />
          </div>,
          <div
            className="event"
          >
            <div
              className="comingEvents"
            >
              M5 rs trip to walsigham
            </div>
            <div
              className="comingEventsDates"
            >
              25 June
            </div>
            <div
              className="time"
            >
              
            </div>
            <div
              className="clear"
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="comingEvents"
              >
                Half terms End
              </div>,
              <div
                className="comingEventsDates"
              >
                01 May
              </div>,
              <div
                className="time"
              >
                
              </div>,
              <div
                className="clear"
              />,
            ],
            "className": "event",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Half terms End",
                "className": "comingEvents",
              },
              "ref": null,
              "rendered": "Half terms End",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "01 May",
                "className": "comingEventsDates",
              },
              "ref": null,
              "rendered": "01 May",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "time",
              },
              "ref": null,
              "rendered": "",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "clear",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="comingEvents"
              >
                NetBall: first vii vs viii
              </div>,
              <div
                className="comingEventsDates"
              >
                20 May
              </div>,
              <div
                className="time"
              >
                2:30 pm
              </div>,
              <div
                className="clear"
              />,
            ],
            "className": "event",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "NetBall: first vii vs viii",
                "className": "comingEvents",
              },
              "ref": null,
              "rendered": "NetBall: first vii vs viii",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "20 May",
                "className": "comingEventsDates",
              },
              "ref": null,
              "rendered": "20 May",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "2:30 pm",
                "className": "time",
              },
              "ref": null,
              "rendered": "2:30 pm",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "clear",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="comingEvents"
              >
                Parent presentation on healthy living
              </div>,
              <div
                className="comingEventsDates"
              >
                03 June
              </div>,
              <div
                className="time"
              >
                12:35 pm
              </div>,
              <div
                className="clear"
              />,
            ],
            "className": "event",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Parent presentation on healthy living",
                "className": "comingEvents",
              },
              "ref": null,
              "rendered": "Parent presentation on healthy living",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "03 June",
                "className": "comingEventsDates",
              },
              "ref": null,
              "rendered": "03 June",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "12:35 pm",
                "className": "time",
              },
              "ref": null,
              "rendered": "12:35 pm",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "clear",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="comingEvents"
              >
                M5 rs trip to walsigham
              </div>,
              <div
                className="comingEventsDates"
              >
                25 June
              </div>,
              <div
                className="time"
              >
                
              </div>,
              <div
                className="clear"
              />,
            ],
            "className": "event",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "M5 rs trip to walsigham",
                "className": "comingEvents",
              },
              "ref": null,
              "rendered": "M5 rs trip to walsigham",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "25 June",
                "className": "comingEventsDates",
              },
              "ref": null,
              "rendered": "25 June",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "time",
              },
              "ref": null,
              "rendered": "",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "clear",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
